File: sample/src/main/java/vipsffm/sample/RawGetVersionSample.java
package vipsffm.sample;

import app.photofox.vipsffm.jextract.VipsRaw;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import vipsffm.RunnableSample;

import java.lang.foreign.Arena;
import java.nio.file.Path;

/**
 * Sense check that version string via VipsRaw works
 */
public class RawGetVersionSample implements RunnableSample {

    private static final Logger logger = LoggerFactory.getLogger(RawGetVersionSample.class);

    @Override
    public void run(Arena arena, Path workingDirectory) {
        var result = VipsRaw.vips_init(arena.allocateFrom("vips-ffm"));
        if (result != 0) {
            throw new RuntimeException("failed to initialise libvips, code " + result);
        }

        var versionPointer = VipsRaw.vips_version_string();
        var vipsVersion = versionPointer == null ? null : versionPointer.getString(0);
        if (vipsVersion == null || vipsVersion.isBlank()) {
            throw new RuntimeException("failed to get libvips version");
        }

        logger.info("libvips version: \"{}\"", vipsVersion);
    }
}

File: sample/src/main/kotlin/vipsffm/RawGetVersionSample.kt
package vipsffm

import app.photofox.vipsffm.generated.VipsRaw
import org.slf4j.LoggerFactory
import java.lang.foreign.Arena
import java.nio.file.Path

object RawGetVersionSample : RunnableSample {

    val logger = LoggerFactory.getLogger(RawGetVersionSample::class.java)

    override fun run(arena: Arena, workingDirectory: Path): Result<Unit> {
        val result = VipsRaw.vips_init(arena.allocateFrom("vips-ffm"))
        if (result != 0) {
            return Result.failure(
                RuntimeException("failed to initialise libvips, code $result")
            )
        }

        val vipsVersion = VipsRaw.vips_version_string()?.getString(0)
        if (vipsVersion.isNullOrBlank()) {
            return Result.failure(
                RuntimeException("failed to get libvips version")
            )
        }

        logger.info("libvips version: \"$vipsVersion\"")
        return Result.success(Unit)
    }
}
